### PR TITLE
Add IntelliJ Idea Migration Map reader and writer

### LINKED ISSUE #61: IntelliJ Idea Migration Map support
Just a basic XML file, with examples being available [here](https://gist.github.com/bonii-xx/dcb1b6a9a1d13b69a1c9), [here](https://gist.github.com/gigaherz/aef4327298473307ae92a6e754fce0d2) and [here](https://gist.github.com/bonii-xx/dcb1b6a9a1d13b69a1c9)

### PR DESCRIPTION
**Pull request by NebelNidas**
Depends on #92 and closes #61.

### CODE REVIEW

**Review by modmuss50 — APPROVED** [review]

File: src/main/java/net/fabricmc/mappingio/format/intellij/MigrationMapFileReader.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2023 FabricMC
```
Comment (liach) [inline src/main/java/net/fabricmc/mappingio/format/intellij/MigrationMapFileReader.java]: Year need an update after reviving this PR after many months.

**Review by liach — COMMENTED** [review]
Didn't spot any problem with migration maps. But the work to ensure tiny has srcNs seems a bit unrelated to migration maps.

File: src/main/java/net/fabricmc/mappingio/format/intellij/MigrationMapFileReader.java (hunk 1 above)
Comment (NebelNidas) [reply src/main/java/net/fabricmc/mappingio/format/intellij/MigrationMapFileReader.java]: Thanks, fixed!

**Comment by NebelNidas** [comment]
> the work to ensure tiny has srcNs seems a bit unrelated to migration maps.

True, I was adding some header validation tests to the migration map format and thought I might do the same for the other applicable formats as well while already being at it. Doesn't quite warrant a separate PR in my opinion.

> Didn't spot any problem with migration maps.

Perfect, LGTM then :)

**Pull request merged**